db/write_batch.cc

```diff
@@ -1223,18 +1475,15 @@ Status WriteBatch::PopSavePoint() {
   return Status::OK();
 }
 
-Status WriteBatch::AssignTimestamp(
-    const Slice& ts, std::function<Status(uint32_t, size_t&)> checker) {
-  TimestampAssigner ts_assigner(prot_info_.get(), std::move(checker), ts);
-  return Iterate(&ts_assigner);
-}
-
-Status WriteBatch::AssignTimestamps(
-    const std::vector<Slice>& ts_list,
-    std::function<Status(uint32_t, size_t&)> checker) {
-  SimpleListTimestampAssigner ts_assigner(prot_info_.get(), std::move(checker),
-                                          ts_list);
-  return Iterate(&ts_assigner);
+Status WriteBatch::UpdateTimestamps(const Slice& ts,
+                                    std::function<size_t(uint32_t)> checker) {
+  TimestampUpdater<decltype(checker)> ts_updater(prot_info_.get(),
```

Comment:
We can ditch the template if `std::function<size_t(uint32_t)>` is the only type it ever gets instantiated with

Reply:
I expect we can add some custom internal checker objects in the future.
plus, I think I should rename `checker` to `ts_sz_func` to match the header file.

Reply:
Sure, just wanted to mention that that would involve either adding new overloads that take the internal checker types, or turning `WriteBatch::UpdateTimestamps` itself into a template (which has certain consequences when it comes to code organization).